src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import * as Yup from 'yup';
+import { TextInput } from 'hds-react';
+import { Field, Form, Formik } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+
+import Grid from '../../../../common/grid/Grid';
+import Select from '../../../../common/select/Select';
+import FormTypeTitle from '../../editModal/FormTypeTitle';
+import {
+  PeriodType,
+  AdditionalProductTaxEnum,
+  ProductServiceType,
+  PriceUnits,
+} from '../../../../@types/__generated__/globalTypes';
+import {
+  getPeriodTKey,
+  getProductTax,
+  getProductServiceTKey,
+  getPriceUnits,
+} from '../../../../common/utils/translations';
+import Button from '../../../../common/button/Button';
+import styles from './additionalServicesModal.module.scss';
+
+const serviceOptions = Object.values(ProductServiceType);
+const taxOptions = Object.values(AdditionalProductTaxEnum);
+const priceUnitOptions = Object.values(PriceUnits);
+// const periodOptions = Object.values(PeriodType);
+
+export interface AdditionalServiceValues {
+  service: ProductServiceType | null;
+  priceValue: number | string;
+  priceUnit: PriceUnits | null;
+  taxPercentage: AdditionalProductTaxEnum | null;
+  period?: PeriodType | null;
+}
+
+export interface AdditionalServicesModalProps {
+  initialValues?: AdditionalServiceValues | null;
+  periodOptions: PeriodType[];
+  onSubmit: (values: AdditionalServiceValues) => void;
+  closeModal: () => void;
+}
+
+export const getAdditionalServicesValidationSchema = (t: TFunction, periodOptions: PeriodType[]) => {
+  return Yup.object().shape({
+    service: Yup.string().oneOf(serviceOptions).required(t('forms.common.errors.required')),
+    priceValue: Yup.number()
+      .positive()
+      .typeError(t('forms.common.errors.numberType'))
+      .required(t('forms.common.errors.required')),
+    priceUnit: Yup.string().oneOf(priceUnitOptions).required(t('forms.common.errors.required')),
+    taxPercentage: Yup.string()
+      .oneOf(taxOptions)
+      .typeError(t('forms.common.errors.numberType'))
+      .required(t('forms.common.errors.required')),
+    period: Yup.string().oneOf(periodOptions).required(t('forms.common.errors.required')),
+  });
+};
+
+const OPTIONAL_SERVICES = [
+  ProductServiceType.SUMMER_STORAGE_FOR_DOCKING_EQUIPMENT,
+  ProductServiceType.SUMMER_STORAGE_FOR_TRAILERS,
+  ProductServiceType.PARKING_PERMIT,
+  ProductServiceType.DINGHY_PLACE,
+];
+
+const AdditionalServicesModal = ({
```

Comment:
This should be called `AdditionalServicesForm`, as the actual Modal is not included

Reply:
Changed.